### ISSUE TITLE
(aws-cdk/core): large context in cdk.context.json causes E2BIG error during synthesis and build

### ISSUE THREAD

**Issue by Alex-Vol-Amz** [aws/aws-cdk#19261]
### What is the problem?

We have a service with a large cellular architecture. As we are migrating to CDK from LPT we need to use the Vpc.fromLookup method at least once per account. The size of the generated cdk.context.json exceeds 300K and will only keep getting larger. At this scale the process fails with E2BIG (environment string too big) when [CDK code in this location](https://github.com/aws/aws-cdk/blob/57ad1e087edef653d672c1426b920b12962f0f0f/packages/aws-cdk/lib/api/cxapp/exec.ts#L47) attempts to pass the stringified contents of cdk.context.json to the spawned subprocess via the CDK_CONTEXT_JSON environment.

Environment strings in Linux are generally limited to 128K size and cannot be changed without recompiling the kernel!

MAX_ARG_STRLEN is a constant defined as PAGESIZE*32 in [include/uapi/linux/binfmts.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/binfmts.h). Its value cannot be changed without recompiling the kernel.

What options do we have to allow using Vpc.fromLookup which is the recommended way to do this and avoid such a roadblock?

### Reproduction Steps

Add sufficient context to your cdk.context.json to exceed the limits, in our case we have a 300K cdk.context.json file. Since this is essentially a hash map of arbitrary values one could fill much of that with garbage random values reach the breaking limit.

Then try to build/synthesize your app which would invoke the CDK_CONTEXT_JSON passing mechanism for exec.

### What did you expect to happen?

Build and synthesize a CDK cloud service.

### What actually happened?

The call will fail with `spawn E2BIG` and process exits with 1.

### CDK CLI Version

1.145.0 (build 9f96380)

### Node.js Version

12

### OS

Amazon Linux 2

### Language

Typescript

**Comment by Alex-Vol-Amz** [comment]
After some discussion with @rix0rrr I arrived at this workaround which is viable until this is fixed in some way.

Changed my initialization code for the application to this:

```
  const app = new cdk.App({
    context: {
      [ECS_REMOVE_DEFAULT_DESIRED_COUNT]: true,
      ...JSON.parse(fs.readFileSync("cdk.static-context.json", "utf8")),
    },
  });
```

The file `cdk.static-context.json` is a literal copy of the generated `cdk.context.json` from an execution of `cdk synth` when less than the totality of my service accounts were synthesized. I was able to incrementally add more of my service cellular accounts to the pipeline definition such that each increment was not adding more resolved keys from lookups in cdk.context.json than the environment size would allow.

With this code in place and an initially empty `cdk.static-context.json` I repeated the process each time adding more of my service accounts to the pipeline definition accumulating more entries in the `cdk.context.json`. I merged the new entries into the `cdk.static-context.json` in each iteration deleting the `cdk.context.json` after the merge.

In the end of this process my new cdk.static-context.json` was over 250K size and the build and synthesis steps worked without any failures.

I expect as new regions and new cellular accounts are added this process will produce incremental changes to cdk.context.json which will be folded into the new cdk.static-context.json file. I am thinking of adding a "beforeExit" handler to auto-merge the newly created entries into the cdk.static-context.json to avoid having to make the manual steps.
